storey/steps/flatten.py

```diff
@@ -0,0 +1,34 @@
+from itertools import chain
+
+from storey import Flow, Event
+from storey.dtypes import _termination_obj
+
+
+class Flatten(Flow):
+    """
+    Flattens sequences of sequences (i.e lists of lists, sets of sets, etc...) into a single sequence, by default
+    casts sequence to list. When setting to_set=True, casts sequence to set.
+    """
+
+    def __init__(self, to_set: bool = False, **kwargs):
+        super().__init__(**kwargs)
+        self.to_set: bool = to_set
+
+    async def _do(self, event):
+        if event is _termination_obj:
+            return await self._do_downstream(_termination_obj)
+        else:
+            if self.to_set:
```

Comment:
Why do we need to convert to a set at all? if a user want's a unique list let him have a set in his event, or use a Map step to do that

Reply:
Yeah you could be right about that, that's just a use case that came to mind. If anyone will need that they will probably can implement `FlattenUnique` or something.